### PR TITLE
Update dependencies 2022-03

### LINKED ISSUE #4230: Switch to `drupal/views_tree:^2.0@alpha` and remove temporary workaround
# Motivation

The maintainers of the `views_tree` finally got around to accepting and merging some changes that were required for D9 compatibility. A new release has been created, so we no longer need my sandbox version of the module.

# Proposed solution

- Remove the entry in the `repositories` section of composer.json for `drupal-pyrello/views_tree`.
- `composer require drupal/views_tree:^2.0@alpha`.

### PR DESCRIPTION
**Pull request by joewhitsitt**
Reviewing a couple of recommended module updates

- photoswipe stable release - https://www.drupal.org/project/photoswipe/releases/3.1.1
- resolves https://github.com/uiowa/uiowa/issues/4230
- closes https://github.com/uiowa/uiowa/pull/4925 address 1.9.0 to [1.10.0](https://www.drupal.org/project/address/releases/8.x-1.10)
- closes https://github.com/uiowa/uiowa/pull/4878 antibot 1.4.0 to [2.0.0](https://www.drupal.org/project/antibot/releases/2.0.0). See additional release notes: [1.5.0](https://www.drupal.org/project/antibot/releases/8.x-1.5)
- closes https://github.com/uiowa/uiowa/pull/4505 entity_usage 2.0.0-beta5 to [2.0.0-beta6](https://www.drupal.org/project/entity_usage/releases/8.x-2.0-beta6)
- closes https://github.com/uiowa/uiowa/pull/4757 imagemagick 3.1.0 to [3.3.0](https://www.drupal.org/project/imagemagick/releases/8.x-3.3). See additional release notes: [3.2.0](https://www.drupal.org/project/imagemagick/releases/8.x-3.2)
- closes https://github.com/uiowa/uiowa/pull/4755 layout_builder_restrictions 2.8.0 to [2.12.0](https://www.drupal.org/project/layout_builder_restrictions/releases/8.x-2.12). See additional release notes: [2.11](https://www.drupal.org/project/layout_builder_restrictions/releases/8.x-2.11), [2.10](https://www.drupal.org/project/layout_builder_restrictions/releases/8.x-2.10), [2.9.0](https://www.drupal.org/project/layout_builder_restrictions/releases/8.x-2.9)
- ~~metatag 1.16.0 to [1.19.0](https://www.drupal.org/project/metatag/releases/8.x-1.19). See additional release notes: [1.18.0](https://www.drupal.org/project/metatag/releases/8.x-1.18), [1.17.0 ](https://www.drupal.org/project/metatag/releases/8.x-1.17)~~ this was already updated https://github.com/uiowa/uiowa/pull/4828
- closes https://github.com/uiowa/uiowa/pull/4607 migrate_plus 5.1.0 to [5.2.0](https://www.drupal.org/project/migrate_plus/releases/8.x-5.2)
- closes https://github.com/uiowa/uiowa/pull/4610 mimemail 1.0.0-alpha3 to [1.0.0-alpha4](https://www.drupal.org/project/mimemail/releases/8.x-1.0-alpha4)
- closes https://github.com/uiowa/uiowa/pull/4899 smart_date 3.4.2 to [3.5.1](https://www.drupal.org/project/smart_date/releases/3.5.1). See additional release notes: [3.5.0](https://www.drupal.org/project/smart_date/releases/3.5.0), [additional beta/release candidates](https://www.drupal.org/project/smart_date/releases), [3.4.3](https://www.drupal.org/project/smart_date/releases/3.4.3)

Avoiding google_analytics and samlauth right now...

## To Test

- Look over the release notes for each module linked above.
- Review https://github.com/uiowa/uiowa#updating-dependencies for testing updates to dependencies locally.
  - Try saving config related to these modules to see if there are config changes not handled by an update hook.
- No db update issues or config import errors
- Sync policy.clas and check /table-of-contents to make sure the views_tree functionality is still functioning.
- Sync grad and check https://grad.uiowa.ddev.site/node/add/thesis_defense to make sure "End" date input is display:none;
- Test a migration against migrate_plus update.

### CODE REVIEW

**Comment by joewhitsitt** [comment]
Looks like a small regression in smart_date. Even though grad thesis defense date field is set to hide end date, the label is still showing.

Commented on a smart_date issue: https://www.drupal.org/project/smart_date/issues/3261072#comment-14436585

**Comment by joewhitsitt** [comment]
@pyrello do you think https://github.com/uiowa/uiowa/pull/4902#issuecomment-1061103499 / https://github.com/uiowa/uiowa/issues/2899 is a blocker for this update? Should we correct it with CSS for now?

**Comment by pyrello** [comment]
> @pyrello do you think https://github.com/uiowa/uiowa/pull/4902#issuecomment-1061103499 / https://github.com/uiowa/uiowa/issues/2899 is a blocker for this update? Should we correct it with CSS for now?

This isn't a blocker. If we can come up with an easy CSS fix until it is fixed upstream, I'm cool with that.

**Comment by pyrello** [comment]
Any chance of adding #4230?

**Comment by joewhitsitt** [comment]
>If we can come up with an easy CSS fix until it is fixed upstream, I'm cool with that.

I think it would have to be thesis defense specific. I will take a crack at it.

>Any chance of adding https://github.com/uiowa/uiowa/issues/4230?

If you tell me what site is using it and any test instructions that would make sense, I can try and work it in.

**Comment by pyrello** [comment]
>> Any chance of adding https://github.com/uiowa/uiowa/issues/4230?

> If you tell me what site is using it and any test instructions that would make sense, I can try and work it in.

I believe that https://policy.clas.uiowa.edu/ is the only site that is using it. I think this is the page that it is being used on: https://policy.clas.uiowa.edu/table-of-contents. Not sure there would be anything to do other than just make sure that page is still rendering. @mark-bennett-uiowa might want to double check this.

**Comment by joewhitsitt** [comment]
Not in a rush to get this one out since it is unplanned work but reviews are welcome!

**Comment by joewhitsitt** [comment]
Going to add https://github.com/uiowa/uiowa/pull/4925 as part of this

**Comment by richardbporter** [comment]
Doing the old switcheroo (checkout main, composer install, enable caching in local settings files, sync sites, hit sites, checkout feature branch, composer install), I was able to produce a WSOD on https://admissions.uiowa.ddev.site/apply showing the following error:
```
The website encountered an unexpected error. Please try again later.
ArgumentCountError: Too few arguments to function Drupal\serialization\Normalizer\DateTimeNormalizer::__construct(), 0 passed in /var/www/html/docroot/core/lib/Drupal/Component/DependencyInjection/Container.php on line 262 and exactly 1 expected in Drupal\serialization\Normalizer\DateTimeNormalizer->__construct() (line 50 of core/modules/serialization/src/Normalizer/DateTimeNormalizer.php).
```

I don't know what update is causing that. I don't see any database updates for that site. Running `drush cim` for that site also throws an error:
```
[error]  TypeError: Argument 2 passed to Drupal\entity_usage\Commands\EntityUsageCommands::__construct() must be an instance of Drupal\entity_usage\EntityUsageQueueBatchManager, instance of Drupal\Core\Entity\EntityTypeManager given
```

Running `drush cr` fixes both errors. Labeling this PR and noting that it probably needs to be run off-hours.

**Comment by joewhitsitt** [comment]
>noting that it probably needs to be run off-hours

converting to draft

**Comment by richardbporter** [comment]
I'm actually seeing that entity_usage error while doing a large `dsa`. It doesn't happen on every site. The AC task will always complete but should report some sites failed. A manual cache rebuild might be required for those sites that failed.

**Comment by joewhitsitt** [comment]
Hmm. Do you have a few sites to try to reproduce this on?

**Comment by richardbporter** [comment]
Seems to happen with with brand and dentistry using the process I outlined above, which is kinda slow going. It might just depend on what is cached, not sure if there is anything specific with these sites.

**Comment by richardbporter** [comment]
Steps for copy/paste:
```
git checkout main
ddev composer install
ddev blt sync --site brand.uiowa.edu
git checkout update_dependencies_202203
ddev composer install
ddev blt auda -vv
```

**Comment by richardbporter** [comment]
> I'm actually seeing that entity_usage error while doing a large `dsa`. It doesn't happen on every site. The AC task will always complete but should report some sites failed. A manual cache rebuild might be required for those sites that failed.

I think the reason this was happening was because our version of `blt auda`, which is what runs during deployments, was not invoking the `drupal:update` process on the site because BLT was returninh FALSE from `Inspector::isDrupalInstalled`. I added a new method to determine this ourselves and started using that. The docs give more details.

TLDR; I think the updb cr should now fix the error.

**Comment by joewhitsitt** [comment]
I just tried the steps above and didn't encounter the error. Thanks for looking into this

**Comment by richardbporter** [comment]
> > I'm actually seeing that entity_usage error while doing a large `dsa`. It doesn't happen on every site. The AC task will always complete but should report some sites failed. A manual cache rebuild might be required for those sites that failed.
> 
> I think the reason this was happening was because our version of `blt auda`, which is what runs during deployments, was not invoking the `drupal:update` process on the site because BLT was returninh FALSE from `Inspector::isDrupalInstalled`. I added a new method to determine this ourselves and started using that. The docs give more details.
> 
> TLDR; I think the updb cr should now fix the error.

Noticing some other issues on another branch related to this. I'm probably going to cherry-pick this change to a hotfix PR.

**Comment by joewhitsitt** [comment]
@GaryRidgway curious how far you got with testing?

**Comment by GaryRidgway** [comment]
only got to the default site before i left for lunch

**Comment by GaryRidgway** [comment]
I reviewed a pretty large amount of this, except for the migration at the end. @cory-skeers , would you be able to assist with this?

**Comment by cory-skeers** [comment]
> I reviewed a pretty large amount of this, except for the migration at the end. @cory-skeers , would you be able to assist with this?

Yup! On it.

**Comment by cory-skeers** [comment]
Checked the migration_plus release notes, and all looked good. Also tested against the uipress_migrate and international_migrate, and both still worked as expected.

**Comment by joewhitsitt** [comment]
Sounds like the plan is to delay this so it doesn't interfere with site launches.

**Comment by joewhitsitt** [comment]
That is a no-go for merge. I repeat. No-go.

**Review by GaryRidgway — APPROVED** [review]
Bang. Approved.
